programs/drift/src/state/oracle.rs

```diff
@@ -231,46 +229,63 @@ pub fn get_pyth_stable_coin_price(
     Ok(oracle_price_data)
 }
 
-// pub fn get_switchboard_price(
-//     _price_oracle: &AccountInfo,
-//     _clock_slot: u64,
-// ) -> DriftResult<OraclePriceData> {
-//     updating solana/anchor cause this to make compiler complan
-//     fix when we're using switchboard again
-//     let aggregator_data = AggregatorAccountData::new(price_oracle)
-//         .or(Err(crate::error::ErrorCode::UnableToLoadOracle))?;
-//
-//     let price = convert_switchboard_decimal(&aggregator_data.latest_confirmed_round.result)?;
-//     let confidence =
-//         convert_switchboard_decimal(&aggregator_data.latest_confirmed_round.std_deviation)?;
-//
-//     // std deviation should always be positive, if we get a negative make it u128::MAX so it's flagged as bad value
-//     let confidence = if confidence < 0 {
-//         u128::MAX
-//     } else {
-//         let price_10bps = price
-//             .unsigned_abs()
-//             .safe_div(1000)
-//             ?;
-//         max(confidence.unsigned_abs(), price_10bps)
-//     };
-//
-//     let delay: i64 = cast_to_i64(clock_slot)?
-//         .safe_sub(cast(
-//             aggregator_data.latest_confirmed_round.round_open_slot,
-//         )?)
-//         ?;
-//
-//     let has_sufficient_number_of_data_points =
-//         aggregator_data.latest_confirmed_round.num_success >= aggregator_data.min_oracle_results;
-//
-//     Ok(OraclePriceData {
-//         price,
-//         confidence,
-//         delay,
-//         has_sufficient_number_of_data_points,
-//     })
-// }
+pub fn get_switchboard_price(
+    price_oracle: &AccountInfo,
+    clock_slot: u64,
+) -> DriftResult<OraclePriceData> {
+    let aggregator_data_loader: AccountLoader<AggregatorAccountData> =
+        AccountLoader::try_from(price_oracle).or(Err(ErrorCode::UnableToLoadOracle))?;
+    let aggregator_data = aggregator_data_loader
+        .load()
+        .or(Err(ErrorCode::UnableToLoadOracle))?;
+
+    let price = convert_switchboard_decimal(&aggregator_data.latest_confirmed_round.result)?
+        .cast::<i64>()?;
+    let confidence =
+        convert_switchboard_decimal(&aggregator_data.latest_confirmed_round.std_deviation)?
+            .cast::<i64>()?;
+
+    // std deviation should always be positive, if we get a negative make it u128::MAX so it's flagged as bad value
+    let confidence = if confidence < 0 {
+        u64::MAX
+    } else {
+        let price_10bps = price.unsigned_abs().safe_div(1000)?;
+        confidence.unsigned_abs().max(price_10bps)
```

Comment:
itchboard_solana/oracle_program/accounts/aggregator/struct.AggregatorAccountData.html#method.check_confidence_interval

perhaps should look into this instead of std_deviation?